src/approveAndCall.ts

```diff
@@ -47,6 +47,42 @@ export abstract class ApproveAndCall {
     }
   }
 
+  public static encodeAddLiquidity(position: Position, addLiquidityOptions: AddLiquidityOptions): string {
```

Comment:
we should probably `Omit` a few things from `AddLiquidityOptions` (`CommonAddLiquidityOptions` more specifically):

- `deadline` because it's assumed that it'll be checked in the swap router `multicall`
- `useNative` because we only are adding WETH here 
- `token{0,1}Permit` because permits need to happen on the swap router not the position manager

we also have to decide what to do with `createPool` from `MintSpecificOptions`. my preference is definitely to `Omit` it as well, even though in theory i believe we could support it by encoding the pool creation.

Reply:
oh wow, didn't know about ts `Omit`, very cool. will omit these for now. can easily add pool creation later.